File: content/ex-mm-daxpy.c
void matmul(double* a, double* b, double* c, long n) {
  for(long i = 0; i < n; i++) {
    for(long j = 0; j < n; j++) a[i*n + j] = 0;
    for(long k = 0; k < n; k++) {
      for(long j = 0; j < n; j++) {
        a[i*n + j] += b[i*n + k] * c[k*n + j];
      }
    }
  }
}

File: content/ex-mm-daxpy.f90
module matmul_module
  use rtvl_module
  implicit none
contains
  subroutine matmul(a, b, c, n)
    integer, intent(in) :: n
    double precision, intent(inout) :: a(rtvl(n),n)
    double precision, intent(in) :: b(rtvl(n),n), c(rtvl(n),n)
    integer :: i, j, k

    do j = 1, n
      a(1:n,j) = 0
      do k = 1, n
        do i = 1, n
          a(i,j) = a(i,j) + b(i,k) * c(k,j)
        end do
      end do
    end do
  end subroutine matmul
end module matmul_module
